models/demos/mistral7b/tt/mistral_model.py

```diff
@@ -0,0 +1,81 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import ttnn
+import torch
+import torch.nn as nn
+from models.demos.mistral7b.tt.mistral_decoder import TtTransformerBlock
+from models.demos.mistral7b.tt.mistral_rms_norm import TtRMSNorm
+import ttnn
+from typing import Optional
+
+
+class TtTransformer(nn.Module):
+    def __init__(
+        self,
+        args,
+        dtype,
+        device,
+        state_dict,
+        weight_cache_path,
+        layers,
+        tt_cos_cached,
+        tt_sin_cached,
+    ):
+        super().__init__()
+        self.args = args
+        self.vocab_size = args.vocab_size
+        self.n_layers = args.n_layers
+        self.device = device
+        assert self.vocab_size > 0
+
+        self.layers = torch.nn.ModuleList(
+            [
+                TtTransformerBlock(
+                    args=args,
+                    device=device,
+                    dtype=dtype,
+                    state_dict=state_dict,
+                    weight_cache_path=weight_cache_path,
+                    layer_num=i,
+                    tt_cos_cached=tt_cos_cached,
+                    tt_sin_cached=tt_sin_cached,
+                )
+                for i in layers
+            ]
+        )
+        self.norm = TtRMSNorm(
+            device=device,
+            state_dict=state_dict,
+            weight_cache_path=weight_cache_path,
+            dtype=dtype,
+            layer_num=None,
+            weight_key="norm",
+        )
+        self.state_dict = state_dict
+
+        self.output_weight = ttnn.as_tensor(
+            self.state_dict["output.weight"].permute(1, 0),
+            device=device,
+            layout=ttnn.TILE_LAYOUT,
+            dtype=dtype,
+            memory_config=ttnn.DRAM_MEMORY_CONFIG,
+            cache_file_name=weight_cache_path / "output.weight",
+        )
+
+    def forward(
+        self,
+        x: ttnn.Tensor,
+        start_pos: int,
+        current_pos: int,
+        attn_masks: Optional[ttnn.Tensor],
+        rot_mat: ttnn.Tensor,
+    ):
+        for layer in self.layers:
+            x = layer(x, start_pos, current_pos, attn_masks, rot_mat)
+
+        x = self.norm(x)
+        output = ttnn.linear(x, self.output_weight, core_grid=ttnn.CoreGrid(x=8, y=8))
```

Comment:
Should we specify instead the grid_size in model_config to something like:

 max_grid_size = (8, 8) if device.arch() == ttl.device.Arch.WORMHOLE_B0 else (9, 12)

Reply:
does this op have to run on max 8, 8 grid? If that's the case, it won't work on any harvested wormhole

Reply:
No. I can reduce to (x=4,y=8) grid. X==rows, right?

Reply:
x is along the row, so it's the number of columns. `(x=4, y=8)` should fail on harvested wormhole. Unless you are using these coords to explicitly set sharding specs, we should use default device compute grid instead of hardcoding like this.

Please change.

Reply:
Pushed the change to pick device's grid size instead.

```
grid_size = device.compute_with_storage_grid_size()
self.max_grid_size = ttnn.CoreGrid(x=grid_size.x, y=grid_size.y)
```

Reply:
this still works for you? If so, good to merge.

FYI, ttnn coordinates are `(y, x)`, so if you want to avoid confusion you should probably flip the coordinates for ttnn. Constructing the coordinates with kwargs is fine as well, but just know that it's actually `(y, x)` by default.